subsys/net/ip/tcp2.c

```diff
@@ -462,7 +462,13 @@ static size_t tcp_data_get(struct tcp *conn, struct net_pkt *pkt)
 
 	if (len > 0) {
 		if (conn->context->recv_cb) {
-			struct net_pkt *up = net_pkt_clone(pkt, K_NO_WAIT);
+			struct net_pkt *up =
+				net_pkt_clone(pkt, TCP_PKT_ALLOC_TIMEOUT);
```

Comment:
Ok for now, but we need to revisit all clone calls in TCP2 at some point and try to replace them by just increasing ref count.